api/src/main/proto/stellarstation/api/v1/radio/radio.proto

```diff
@@ -51,11 +51,125 @@ message CommunicationProtocol {
   oneof Framing {
     // AX.25 protocol settings.
     AX25 ax25 = 1;
+
+    // CCSDS framing settings.
+    CCSDS ccsds = 2;
+
+    // Bit stream framing settings.
+    BitStream bit_stream = 3;
   }
 }
 
-// Communication of a stream of bits with no framing defined.
+// CCSDS framing https://public.ccsds.org/Pubs/131x0b3e1.pdf
+// It allows for a few common variations satellites use such
+// as setting a custom synchronization word, inverting the
+// convolutional coding polynomials, etc. The sequence of
+// error-correcting codes will be applied as they are defined
+// in the CCSDS standard.
+message CCSDS {
+  // Synchronization word parameters. Although CCSDS defines
+  // a particular synchronization word, this API lets the user
+  // select a custom one.
+  SynchronizationParams synchronization_params = 1;
+
+  // CCSDS-specific convolutional coding parameters.
+  // Code rate (r): 1/2
+  // Constraint length (K): 7
+  // Connection vectors:
+  //   G1 = 1111001
+  //   G2 = 1011011
+  message ConvolutionalCodingParams {
+    // Whether or not convolutional coding is applied
+    bool enabled = 1;
+
+    // Whether or not the G1 polynomial is inverted
+    bool g1_inverted = 2;
+
+    // Whether or not the G2 polynomial is inverted
+    bool g2_inverted = 3;
+
+    // Enum describing different rates for puncturing
+    enum Puncturing {
+      // Puncturing unused
+      OFF = 0;
+
+      // Puncturing rate 2/3
+      PUNCTURE_2_3 = 1;
+
+      // Puncturing rate 3/4
+      PUNCTURE_3_4 = 2;
+
+      // Puncturing rate 4/5
+      PUNCTURE_4_5 = 3;
+
+      // Puncturing rate 5/6
+      PUNCTURE_5_6 = 4;
+
+      // Puncturing rate 6/7
+      PUNCTURE_6_7 = 5;
+
+      // Puncturing rate 7/8
+      PUNCTURE_7_8 = 6;
+    }
+
+    // Optional puncturing for convolutional code
+    Puncturing puncturing = 4;
+  }
+
+  // Convolutional coding parameters
+  ConvolutionalCodingParams convolutional_coding_params = 2;
+
+  // CCSDS-specific scrambling parameters.
+  // Additive scrambler with pseudo-random
+  // sequence generated by the polynomial:
+  // h(x) = x^8 + x^7 + x^5 + x^3 + 1
+  message ScramblingParams {
```

Comment:
CCSDSScramblingParams

Reply:
I think it's common, it's not a hard recommendation though. The main reason it came to mind is to match the message name to it's primary comment - CCSDS-specific scrambling parameters are CCSDSScramblingParams.